src/main/java/com/dinegood/springbootproject/controller/CartController.java

```diff
@@ -0,0 +1,52 @@
+package com.dinegood.springbootproject.controller;
+
+import com.dinegood.springbootproject.model.CartDetails;
+import com.dinegood.springbootproject.services.CartService;
+import com.dinegood.springbootproject.model.Itemdetails;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.Optional;
+
+@RestController
+@RequestMapping("/cart")
+public class CartController {
+    private final CartService cartService;
+
+    @Autowired
+    public CartController(CartService cartService) {
+        this.cartService = cartService;
+    }
+
+
```

Comment:
Applies to all Controller methods:
- don't keep null check on entity, all exceptions must be caught and inside catch we need to return bad request or equivalent response entity.

Reply:
Null check is being used to return success as well as the failure message through response entity I don't think using try catch block can serve the success message as it require to return the entity to throw an exception.